utf8.c

```diff
@@ -259,11 +262,56 @@ vfmprintf(FILE *stream, const char *fmt, va_list ap)
 {
 	char	*str;
 	int	 ret;
-
+		
 	if ((ret = vasnmprintf(&str, INT_MAX, NULL, fmt, ap)) < 0)
 		return -1;
+#ifdef WINDOWS
```

Comment:
Please move this functionality to posix layer and reduce the use of #ifdef windows..